Source/WebCore/history/CachedPage.cpp

```diff
@@ -144,10 +141,11 @@ void CachedPage::restore(Page& page)
 #if PLATFORM(IOS_FAMILY)
         // We don't want focused nodes changing scroll position when restoring from the cache
         // as it can cause ugly jumps before we manage to restore the cached position.
-        localMainFrame->selection().suppressScrolling();
+        if (localMainFrame)
+            localMainFrame->selection().suppressScrolling();
 
         bool hadProhibitsScrolling = false;
-        auto* frameView = localMainFrame->view();
+        auto* frameView = localMainFrame ? localMainFrame->view() : nullptr;
```

Comment:
The only function used below is in ScrollView, which is in all FrameViews local or remote. Is there a reason we can't use `auto* frameView = mainFrame.view();` on line 148 instead?